src/node_buffer.cc

```diff
@@ -97,11 +141,18 @@ Local<Object> New(Isolate* isolate, Handle<String> string, enum encoding enc) {
   EscapableHandleScope scope(isolate);
 
   size_t length = StringBytes::Size(isolate, string, enc);
+  char* data = static_cast<char*>(malloc(length));
+
+  if (data == nullptr)
+    return scope.Escape(Local<Object>());
+
+  size_t actual = StringBytes::Write(isolate, data, length, string, enc);
+  CHECK(actual <= length);
 
-  Local<Object> buf = New(isolate, length);
-  char* data = Buffer::Data(buf);
-  StringBytes::Write(isolate, data, length, string, enc);
+  if (actual < length)
+    data = static_cast<char*>(realloc(data, actual));
```

Comment:
There is no plausible scenario where this can fail but maybe add a `CHECK_NE(data, nullptr)` anyway.